spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonServerAutoConfiguration.java

```diff
@@ -259,6 +264,62 @@ public PersistentStreamMessageSourceRegistrar persistentStreamRegistrar(
         return new PersistentStreamMessageSourceRegistrar(environment, executorBuilder);
     }
 
+    @Configuration
+    @ConditionalOnProperty(name = "axon.axonserver.auto-persistent-streams.enabled")
+    public static class AutoPersistentStreamConfiguration {
+
+        /**
+         * Creates {@link AxonServerConfiguration.PersistentStreamSettings} bean for auto persistent streams.
+         *
+         * @return {@link AxonServerConfiguration.PersistentStreamSettings}
+         */
+        @Bean
+        @ConfigurationProperties(prefix = "axon.axonserver.auto-persistent-streams")
+        AxonServerConfiguration.PersistentStreamSettings autoPersistentStreamSettings() {
+            return new AxonServerConfiguration.PersistentStreamSettings();
+        }
+    }
+
+    /**
+     * The method registers {@link EventProcessingConfigurer.SubscribableMessageSourceDefinitionBuilder} in
+     * {@link EventProcessingModule}
+     *
+     * @param executorBuilder       is used to create
+     *                              {@link EventProcessingConfigurer.SubscribableMessageSourceDefinitionBuilder}
+     * @param eventProcessingModule where {@link EventProcessingConfigurer.SubscribableMessageSourceDefinitionBuilder}
+     *                              will be registered.
+     * @param psFactory             is used to create
+     *                              {@link EventProcessingConfigurer.SubscribableMessageSourceDefinitionBuilder}
+     * @return
+     */
+    @Bean
+    @ConditionalOnProperty(name = "axon.axonserver.auto-persistent-streams.enabled")
+    public EventProcessingConfigurer.SubscribableMessageSourceDefinitionBuilder autoPersistentStreamMessageSourceDefinitionBuilder(
```

Comment:
in fact the method don't have to return anything, but I couldn't find a way to make spring to invoke this method. I tried to put @Autowire instead of @Bean, but it doesn't work here. Any suggestion?

Reply:
Yes, I do! You can add a `ConfigurerModule` bean to the Spring Application Context. The `ConfigurerModule` is a lambda consuming the entire `Configurer` for you. Hence, it allows you to add register operations.

Axon Framework will automatically look for all `ConfigurerModule` beans and invoke those.